.circleci/config.yml

```diff
@@ -4,49 +4,69 @@
 #
 version: 2
 jobs:
-  build:
+  ui:
+    machine:
+      image: circleci/classic:latest
+    steps:
+      - checkout
+      - restore_cache:
+          key: node-modules-{{ .Branch }}--{{ checksum "ui/package.json" }}
```

Comment:
I'd include the secondary key here and below as well (with just the branch); this will mean quicker cached builds when we change package.json.

Reply:
Per offline discussion, this could cause issues when removing / downgrading a dependency.